lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -89,35 +103,104 @@ bool SwiftUnsafeBufferPointer::Update() {
   if (CompilerType type = some_value_sp->GetCompilerType())
     argument_type = SwiftASTContext::GetGenericArgumentType(type, 0);
 
-  if (!argument_type.IsValid())
-    return false;
-
-  m_elem_type = argument_type;
+  if (argument_type.IsValid())
```

Comment:
`UnsafeRawBufferPointer` doesn't have a generic argument since by design, it's a raw byte view of the pointed object. Since `GetAddress` is used for both type, when it's called from the `UnsafeRawBufferPointer` helper class, the generic argument type will be invalid, which leaves `m_elem_type` uninitialized (invalid). Later on, in `UnsafeRawBufferPointer::Update`, I check if `m_elem_type` is invalid, in which case, I reconstruct the `Swift.UInt8` compiler type.

If you look at `UnsafeRawBufferPointer`'s layout, we can't get the generic argument type with the top-level ValueObject ("_position"). To get it, lldb need to look one level below, at the "some" ValueObject node.
That's the reason why I do it in `GetAddress`, on the way of fetching the "_pointerValue" address.

I should probably add a comment to explain this.